main.go

```diff
@@ -77,12 +76,31 @@ type encryptCloudImageBase struct {
 
 	workingDir string
 	devPath    string
+	devPathFormat string
 
 	partitions    gpt.Partitions
 	rootPartition *gpt.PartitionEntry
 	esp           *gpt.PartitionEntry
 }
 
+func (b *encryptCloudImageBase) initDevPathFormat(devPath string) error {
+	b.devPath = devPath
+
+	if strings.HasPrefix(b.devPath, "/dev/nbd") || strings.HasPrefix(b.devPath, "/dev/nvme") {
+		b.devPathFormat = "%sp%d"
+	} else if strings.HasPrefix(b.devPath, "/dev/sd") || strings.HasPrefix(b.devPath, "/dev/vd") {
+		b.devPathFormat = "%s%d"
```

Comment:
Rather than setting the format string here, can we just have a static format string but configure a prefix instead (eg `/dev/nbdp`, `/dev/nvmen0p`, `/dev/vda1`?

Reply:
I wanted to keep the functions generating the path similar to the original code. If you think this really needs to be changed, please elaborate in more details what prefix to configure - I'm not clear on that (i.e. not sure what /dev/nbdp mean since the input is /dev/nbd5 for instance which needs to be come /dev/nbd5p1).